### PR TITLE
[image] Omit `children` in the props type

### PR DESCRIPTION
**Pull request by ashaller2017**
Omit children from ImageProps

# Why

Want to have type hinting when including children in Image props. Should use ImageBackground when needing children. 

# How

Update the ImageProps type to omit children.

# Test Plan

Created fresh expo project importing my image type. Type hints when trying to pass children and doesn't when self closing. 

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-image/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,12 @@
 
 ### 💡 Others
 
+## 2.0.3 — 2024-11-23
+
+- Update `ImageProps` so `children` are omitted. ([#33afe30](https://github.com/expo/expo/pull/33210) by [@ashaller2017](https://github.com/ashaller2017)) ([#33210](https://github.com/expo/expo/pull/33210) by [@ashaller2017](https://github.com/ashaller2017))
+
+### 💡 Others
+
```
Comment (tsapeta) [inline packages/expo-image/CHANGELOG.md:18]: Changelog entries should be put under "Unpublished" section. Version and the date is automatically added by the publish script.
```suggestion
- Update `ImageProps` type so `children` are omitted. ([#33210](https://github.com/expo/expo/pull/33210) by [@ashaller2017](https://github.com/ashaller2017))

```

File: packages/expo-image/src/Image.types.ts [hunk 2]
```diff
@@ -89,7 +89,7 @@ export type ImageDecodeFormat = 'argb' | 'rgb';
  * Some props are from React Native Image that Expo Image supports (more or less) for easier migration,
  * but all of them are deprecated and might be removed in the future.
  */
-export interface ImageProps extends Omit<ViewProps, 'style'> {
+export interface ImageProps extends Omit<ViewProps, 'style' | 'children'> {
```
Comment (tsapeta) [inline packages/expo-image/src/Image.types.ts:92]: As this changes types, you also need to run `yarn build` in `packages/expo-image` to rebuild TS definitions in the build folder

**Review by tsapeta — CHANGES_REQUESTED** [review]
Thanks! Left two comments below.

File: packages/expo-image/src/Image.types.ts (hunk 2 above)
Comment (ashaller2017) [reply packages/expo-image/src/Image.types.ts:92]: Ok on it! Thank you.

File: packages/expo-image/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,8 @@
 
 ## Unpublished
 
+- Update `ImageProps` type so `children` are omitted. ([#33210](https://github.com/expo/expo/pull/33210) by [@ashaller2017](https://github.com/ashaller2017))
```
Comment (tsapeta) [inline packages/expo-image/CHANGELOG.md:5]: It's still in a wrong place. Can you please move it to "Others" category?

File: packages/expo-image/src/Image.tsx [hunk 4]
```diff
@@ -199,13 +199,6 @@ export class Image extends React.PureComponent<ImageProps> {
       loggedDefaultSourceDeprecationWarning = true;
     }
 
-    if (restProps.children && !loggedRenderingChildrenWarning) {
-      console.warn(
-        'The <Image> component does not support children. If you want to render content on top of the image, consider using the <ImageBackground> component or absolute positioning.'
-      );
-      loggedRenderingChildrenWarning = true;
-    }
```
Comment (tsapeta) [inline packages/expo-image/src/Image.tsx:207]: I think we don't want to remove that warning – it's good to let the user know if he is doing something wrong

**Review by tsapeta — CHANGES_REQUESTED** [review]

File: packages/expo-image/src/Image.tsx (hunk 4 above)
Comment (ashaller2017) [reply packages/expo-image/src/Image.tsx:207]: When building Expo-Image, Omitting the children will cause a build failure. This initial logger was setup due to a discord discussion we had when it wasn't clear you could not wrap children in an image component. By type hinting this error should be apparent. I could rebuild throwing a @ts-ignore to build this. What do you think?

File: packages/expo-image/CHANGELOG.md (hunk 3 above)
Comment (ashaller2017) [reply packages/expo-image/CHANGELOG.md]: Read previous comment wrong. Apologies.

File: packages/expo-image/src/Image.tsx (hunk 4 above)
Comment (tsapeta) [reply packages/expo-image/src/Image.tsx:207]: In development Metro bundler will not complain if you have issues in types, it strips out all the types. Your code editor will hint you about the mismatch in types, but I think this is not enough. Especially when your app is big, you don't immediately see all such issues when you upgrade the package until the component is used.

File: packages/expo-image/src/Image.tsx [hunk 5]
```diff
@@ -198,14 +198,14 @@ export class Image extends React.PureComponent<ImageProps> {
...
-
+    // @ts-expect-error
     if (restProps.children && !loggedRenderingChildrenWarning) {
       console.warn(
         'The <Image> component does not support children. If you want to render content on top of the image, consider using the <ImageBackground> component or absolute positioning.'
       );
       loggedRenderingChildrenWarning = true;
     }
-
+
```
Comment (tsapeta) [inline packages/expo-image/src/Image.tsx:208]: Linter is failing on this 😉 Don't forget to rebuild files after this is applied (not sure if this is needed for just whitespace changes, probably not).
```suggestion

```

**Pull request merged**